.github/workflows/advanced.yml

```diff
@@ -427,6 +431,62 @@ jobs:
           path: ${{ env.ARTIFACT_PATH }}/${{ env.TOOLCHAIN_NAME }}-zlib.zip
           retention-days: 3
 
+  build-libxml2:
+    needs: [build-toolchain, build-zlib]
+    runs-on: ubuntu-latest
+
+    env:
+      ZLIB_PATH: ${{ github.workspace }}/zlib
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          path: ${{ github.workspace }}
+
+      - name: Checkout libxml2
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.LIBXML2_REPO }}
+          ref: ${{ env.LIBXML2_BRANCH }}
+          path: ${{ env.SOURCE_PATH }}/${{ env.LIBXML2_VERSION }}
+
+      - name: Download toolchain
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ env.ARTIFACT_PATH }}/${{ env.TOOLCHAIN_PACKAGE_NAME }}
+          key: ${{ env.TOOLCHAIN_NAME }}-${{ needs.build-toolchain.outputs.workflow-sha }}-${{ needs.build-toolchain.outputs.binutils-sha }}-${{ needs.build-toolchain.outputs.gcc-sha }}-${{ needs.build-toolchain.outputs.mingw-sha }}-${{ needs.build-toolchain.outputs.binutils-scripts-sha }}-${{ needs.build-toolchain.outputs.toolchain-scripts-sha }}
```

Comment:
This hash key is used in many places and looks complicated to review, could it be precalculated at one place?

Reply:
The key creation was moved to the previous job output declaration. The caveat is that the `TOOLCHAIN_NAME` needs to be hardcoded, otherwise the latest executed value, e.g. `x86_64-w64-mingw32-ucrt` is set.